sparkmagic/sparkmagic/controllerwidget/addendpointwidget.py

```diff
@@ -59,3 +61,12 @@ def run(self):
         # We need to call the refresh method because drop down in Tab 2 for endpoints wouldn't refresh with the new
         # value otherwise.
         self.refresh_method()
+
+    def update_endpoint(self):
```

Comment:
Let's rename this method to `_show_correct_endpoint_fields` and I believe that its signature should be: `self, name, old_value, new_value` from looking at other callbacks: https://github.com/jupyter-incubator/sparkmagic/blob/master/autovizwidget/autovizwidget/widget/encodingwidget.py#L115